Mathlib/Util/CountHeartbeats.lean

```diff
@@ -176,3 +175,58 @@ elab "count_heartbeats! " n:(num)? "in" ppLine cmd:command : command => do
 end CountHeartbeats
 
 end Mathlib
+
+/-!
+#  The "countHeartbeats" linter
+
+The "countHeartbeats" linter counts the hearbeats of every declaration.
+-/
+
+open Lean Elab Command
+
+namespace Mathlib.Linter
+
+/--
+The "countHeartbeats" linter counts the hearbeats of every declaration.
+
+The effect of the linter is similar to `count_heartbeats in xxx`, except that it applies
+to all declarations.
+
+Note that the linter only counts heartbeats in "top-level" declarations:
+it looks inside `set_option ... in`, but not, for instance, inside `mutual` blocks.
+
+There is a convenience notation `count_heartbeats` that simply sets the linter option to true.
+-/
+register_option linter.countHeartbeats : Bool := {
+  defValue := false
+  descr := "enable the countHeartbeats linter"
+}
+
+namespace CountHeartbeats
+
+@[inherit_doc Mathlib.Linter.linter.countHeartbeats]
+def countHeartbeatsLinter : Linter where run := withSetOptionIn fun stx ↦ do
+  unless Linter.getLinterValue linter.countHeartbeats (← getOptions) do
+    return
+  if (← get).messages.hasErrors then
+    return
+  let mut msgs := #[]
+  if [``Lean.Parser.Command.declaration, `lemma].contains stx.getKind then
+    let s ← get
+    elabCommand (← `(command| count_heartbeats in $(⟨stx⟩)))
+    msgs := (← get).messages.unreported.toArray.filter (·.severity != .error)
+    set s
+  let (ref, declStr) := match stx.find? (·.isOfKind ``Parser.Command.declId) with
+    | some decl => (decl, s!"'{decl[0].getId}' ")
+    | none => (stx, "")
+  for msg in msgs do logInfoAt ref m!"{declStr}{← msg.toString}"
```

Comment:
```suggestion
  match stx.find? (·.isOfKind ``Parser.Command.declId) with
    | some decl =>
      for msg in msgs do logInfoAt decl m!"'{decl[0].getId}' {(← msg.toString).decapitalize}"
    | none =>
      for msg in msgs do logInfoAt stx m!"{← msg.toString}"
```

how about this in order to use `String.decapitalize`?

Reply:
Thanks for the suggestion: I did not know about `decapitalize`.